pkg/controllers/vdb/createdb_reconciler.go

```diff
@@ -199,7 +202,27 @@ func (c *CreateDBReconciler) generatePostDBCreateSQL(ctx context.Context, initia
 }
 
 // postCmdCleanup will handle any cleanup action after initializing the database
-func (c *CreateDBReconciler) postCmdCleanup(_ context.Context) (ctrl.Result, error) {
+func (c *CreateDBReconciler) postCmdCleanup(ctx context.Context) (ctrl.Result, error) {
+	pf, ok := c.findPodToRunInit()
+	if !ok {
+		return ctrl.Result{}, errors.New("could not find a PodFact for create db's post cmd cleanup")
+	}
+	// The generation of httpstls.json is influenced by the DBInitialized status
+	// condition. Now that has changed, we need to set an annotation to continue
+	// getting the same behavior. Since the default behavior is to generate the
+	// file, we need to set an annotation if we didn't generate the file yet.
```

Comment:
For some reason this last part is a little bit confusing to me.

Reply:
Once we have set the DBInitialized status then the function IsHTTPSTlsConfGenerationEnabled() will return true. It doesn't know what version was used to create the database, so it defaults to creating the httpstls.json file. This annotation allows us to keep skipping the generation of that file. It's not perfect because you could to a revive in the future, and if you don't set this annotation, then we will generate the file. But I think its better to generate the file and not need it than the opposite.